scripts/partition_manager.py

```diff
@@ -58,6 +58,12 @@ def remove_irrelevant_requirements(reqs):
     for x in to_delete:
         del reqs[x]
 
+    # Remove 'placement' rules which are superfluous, take the first non-empty one.
+    for k, v in reqs.items():
+        if 'placement' in v.keys() and len(v['placement']) > 1:
+            for ktd in list(v['placement'].keys())[1:]:
```

Comment:
Since dicts are unordered, this keeps a random entry. Maybe it should fail instead?

Reply:
Are they unsorted? AFAICS when giving a yaml file with the keys in a certain order, that order is always kept in python.

Reply:
I stand corrected. For Python 3.7, order is preserved :).

> the insertion-order preservation nature of dict objects has been declared to be an official part of the Python language spec.

https://docs.python.org/3/whatsnew/3.7.html

Reply:
Hm, but only 3.4 is required AFAIK.
By their wording I assume that the order is actually preserved in 3.4 as well.

Reply:
Can a collections.OrderedDict be used?

Reply:
The dict comes from inside pyyaml, but it appears it might still be possible to change the type. https://stackoverflow.com/questions/5121931/in-python-how-can-you-load-yaml-mappings-as-ordereddicts